pkg/gateway/model_build_rule.go

```diff
@@ -34,151 +35,154 @@ const (
 func (t *latticeServiceModelBuildTask) buildRules(ctx context.Context) error {
 
 	var ruleID = 1
-	for _, parentRef := range t.httpRoute.Spec.ParentRefs {
-		if parentRef.Name != t.httpRoute.Spec.ParentRefs[0].Name {
+	for _, parentRef := range t.route.GetSpec().GetParentRefs() {
+		if parentRef.Name != t.route.GetSpec().GetParentRefs()[0].Name {
 			// when a service is associate to multiple service network(s), all listener config MUST be same
 			// so here we are only using the 1st gateway
 			glog.V(2).Infof("Ignore parentref of different gateway %v", parentRef.Name)
 
 			continue
 		}
-		port, protocol, _, err := t.extractListnerInfo(ctx, parentRef)
+		port, protocol, _, err := t.extractListenerInfo(ctx, parentRef)
 
 		if err != nil {
 			glog.V(2).Infof("Error on buildRules %v \n", err)
 			return err
 		}
 
-		for _, httpRule := range t.httpRoute.Spec.Rules {
-			glog.V(6).Infof("Parsing http rule spec: %v\n", httpRule)
+		for _, rule := range t.route.GetSpec().GetRules() {
+			glog.V(6).Infof("Parsing http rule spec: %v\n", rule)
 			var ruleSpec latticemodel.RuleSpec
 
-			if len(httpRule.Matches) > 1 {
+			if len(rule.GetMatches()) > 1 {
 				// only support 1 match today
-				glog.V(2).Infof("Do not support multiple matches: matches == %v ", len(httpRule.Matches))
+				glog.V(2).Infof("Do not support multiple matches: matches == %v ", len(rule.GetMatches()))
 				return errors.New(LATTICE_NO_SUPPORT_FOR_MULTIPLE_MATCHES)
 			}
 
-			if len(httpRule.Matches) == 0 {
+			if len(rule.GetMatches()) == 0 {
 				glog.V(6).Infof("Continue next rule, no matches specified in current rule")
 				continue
 			}
 
 			// only support 1 match today
-			match := httpRule.Matches[0]
-
-			if match.Path != nil && match.Path.Type != nil {
-				glog.V(6).Infof("Examing pathmatch type %v value %v for for httproute %s namespace %s ",
-					*match.Path.Type, *match.Path.Value, t.httpRoute.Name, t.httpRoute.Namespace)
-
-				switch *match.Path.Type {
-				case gateway_api.PathMatchExact:
-					glog.V(6).Infof("Using PathMatchExact for httproute %s namespace %s ",
-						t.httpRoute.Name, t.httpRoute.Namespace)
-					ruleSpec.PathMatchExact = true
-
-				case gateway_api.PathMatchPathPrefix:
-					glog.V(6).Infof("Using PathMatchPathPrefix for httproute %s namespace %s ",
-						t.httpRoute.Name, t.httpRoute.Namespace)
-					ruleSpec.PathMatchPrefix = true
-				default:
-					glog.V(2).Infof("Unsupported path match type %v for httproute %s namespace %s",
-						*match.Path.Type, t.httpRoute.Name, t.httpRoute.Namespace)
-					return errors.New(LATTICE_UNSUPPORTED_PATH_MATCH_TYPE)
+			match := rule.GetMatches()[0]
+
+			if httpMatch, ok := match.(core.HTTPRouteMatch); ok {
```

Comment:
please use type assertion switch, also return error when type does not match:

an example https://go.dev/tour/methods/16

Reply:
Fixed!